fake_stream.sh

```diff
@@ -1,20 +1,231 @@
-if ! sudo modprobe v4l2loopback exclusive_caps=1 card_label="HP webcam"; then
-   echo "Unable to probe kernel module."
-   exit ;
-fi
-
-WEBCAMS=$(ls /dev/video*)
-echo $WEBCAMS
-read -p "Choose Webcam ID (last digit):" ID
-WEBCAM=$(grep $ID <<< $WEBCAMS)
-
-while [ true ]; do
-   read -t 1 -n 1
-   if [ $? = 0 ] ; then
-      # cleanup
-      sudo modprobe -r v4l2loopback
-      exit ;
-   else
-      ffmpeg -re -i $1 -map 0:v -f v4l2 $WEBCAM
-   fi
-done
+#!/bin/bash
+
+# Capture SIGINT and exit gracefully
+trap exitgracefully SIGINT
+
+#Specify Color Schemes
+NONE='\033[00m'
+RED='\033[01;31m'
+GREEN='\033[01;32m'
+YELLOW='\033[01;33m'
+BLUE='\033[01;34m'
+PURPLE='\033[01;35m'
+CYAN='\033[01;36m'
+WHITE='\033[01;37m'
+BOLD='\033[1m'
+BLINK='\033[5m'
+UNDERLINE='\033[4m'
+
+#Set Dimensions For Whiptail
+declare -i L=10
+declare -i W=30
+
+#Global Variables
+VIDEO_STREAM=""
+VIDEO=""
+SOURCE_FLAG=""
+
+#Print Center Banner
+center()
+{
+	local terminal_width=$(tput cols)     # query the Terminfo database: number of columns
+	local text="${1:?}"                   # text to center
+	local glyph="${2:-=}"                 # glyph to compose the border
+	local padding="${3:-2}"               # spacing around the text
+
+	local text_width=${#text}
+
+	local border_width=$(( (terminal_width - (padding * 2) - text_width) / 2 ))
+
+	local border=                         # shape of the border
+
+	# create the border (left side or right side)
+	for ((i=0; i<border_width; i++))
+	do
+		border+="${glyph}"
+	done
+
+	# a side of the border may be longer (e.g. the right border)
+	if (( ( terminal_width - ( padding * 2 ) - text_width ) % 2 == 0 ))
+	then
+		# the left and right borders have the same width
+		local left_border=$border
+		local right_border=$left_border
+	else
+		# the right border has one more character than the left border
+		# the text is aligned leftmost
+		local left_border=$border
+		local right_border="${border}${glyph}"
+	fi
+
+	# space between the text and borders
+	local spacing=
+
+	for ((i=0; i<$padding; i++))
+	do
+		spacing+=" "
+	done
+
+	# displays the text in the center of the screen, surrounded by borders.
+	printf "${bcolor}${left_border}${spacing}${CYAN}${text}${NONE}${spacing}${right_border}\n"
+}
+
+help()
+{
+	echo -e """[$CYAN>$NONE] Usage: $GREEN./fake_stream.sh $NONE[OPTIONS]
+
+OPTIONS :
+-h, --help            Show This Menu
+-s, --source          Stream Desktop
+-v, --video$CYAN FILENAME$NONE  Path to Video/Image
+"""
```

Comment:
I prefer the usage where you just provide the file path for the video stream without any additional flags.
It will be more convenient and backwards compatible to accept video file paths by default. (Not forcing the user to use `-v` flag).